native/Avalonia.Native/src/OSX/Screens.mm

```diff
@@ -43,14 +66,40 @@ virtual HRESULT GetScreen (int index, AvnScreen* ret) override
             
             ret->Scaling = 1;
             
-            ret->IsPrimary = index == 0;
-            
+            ret->IsPrimary = CGDisplayIsMain(displayId);
+
+            // Compute natural orientation:
+            auto naturalScreenSize = CGDisplayScreenSize(displayId);
+            auto isNaturalLandscape = naturalScreenSize.width > naturalScreenSize.height;
+            // Normalize rotation:
+            auto rotation = (int)CGDisplayRotation(displayId) % 360;
+            if (rotation < 0) rotation = 360 - rotation;
+            // Get current orientation relative to the natural
+            if (rotation >= 0 && rotation < 90) {
+                ret->Orientation = isNaturalLandscape ? AvnScreenOrientation::Landscape : AvnScreenOrientation::Portrait;
+            } else if (rotation >= 90 && rotation < 180) {
+                ret->Orientation = isNaturalLandscape ? AvnScreenOrientation::Portrait : AvnScreenOrientation::Landscape;
+            } else if (rotation >= 180 && rotation < 270) {
+                ret->Orientation = isNaturalLandscape ? AvnScreenOrientation::LandscapeFlipped : AvnScreenOrientation::PortraitFlipped;
+            } else {
+                ret->Orientation = isNaturalLandscape ? AvnScreenOrientation::PortraitFlipped : AvnScreenOrientation::LandscapeFlipped;
+            }
+
             return S_OK;
         }
     }
+
+private:
+    static void CGDisplayReconfigurationCallBack(CGDirectDisplayID display, CGDisplayChangeSummaryFlags flags, void *userInfo)
+    {
+        auto events = (IAvnScreenEvents*)userInfo;
```

Comment:
I didn't use any ComPtr here, and just directly passed IAvnScreenEvents* without changing any ownership. Should be fine, I assume, since IAvnScreenEvents owner is C# class that lives whole app execution.

Reply:
Done